### ISSUE TITLE
Fail to install mise v2024.6.2

### ISSUE THREAD

**Issue by MontakOleg** [jdx/mise#2264]
**Describe the bug**
Mise `v2024.6.2` can't be installed via `mise self-update`.

**To Reproduce**
Run `mise self-update`

```
mise self-update
Checking target-arch... mise-v2024.6.2-macos-arm64.tar.gz
Checking current version... v2024.6.1
Checking latest released version... v2024.6.2
New release found! v2024.6.1 --> v2024.6.2
New release is compatible
mise ReleaseError: No asset found for target: `mise-v2024.6.2-macos-arm64.tar.gz`
mise Run with --verbose or MISE_VERBOSE=1 for more information
```

**Expected behavior**
Mise can be updated.

**Additional context**

There is no binaries in GitHub release https://github.com/jdx/mise/releases/tag/v2024.6.2 for some reason:
<img width="1289" alt="image" src="https://github.com/jdx/mise/assets/1800899/d173dfc1-652e-4c3d-b7ca-18dcbe6feee2">

**Comment by yhakbar** [comment]
I think this is likely an issue with CI failing to compile and upload the relevant assets to the release.

If I could make a recommendation:
Assuming that `mise self-update` uses the `/releases/latest` API to discover new releases, a good operational change might be to cut new releases as pre-releases and only set them to the latest, non-pre-release if assets are uploaded successfully.

**Comment by yhakbar** [comment]
Looks like it's fixed for `v2024.6.2`! Thank you maintainer that fixed it!

**Comment by MontakOleg** [comment]
Confirm, fixed, thanks!
